examples/chip-tool/commands/clusters/ModelCommand.cpp

```diff
@@ -37,39 +37,28 @@ void DispatchSingleClusterCommand(chip::ClusterId aClusterId, chip::CommandId aC
         "Default DispatchSingleClusterCommand is called, this should be replaced by actual dispatched for cluster commands");
 }
 
-CHIP_ERROR ModelCommand::Run(PersistentStorage & storage, NodeId localId, NodeId remoteId)
+CHIP_ERROR ModelCommand::Run(NodeId localId, NodeId remoteId)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
-    chip::Controller::CommissionerInitParams initParams;
-    initParams.storageDelegate = &storage;
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
```

Comment:
With this change, does this means ModelCommand::Run have to be called from application thread in future?  If the application plan to use the single thread to drive CHIP stack, there will be double call to chip::DeviceLayer::PlatformMgr().LockChipStack() from CHIP main thread

Reply:
This function is very much designed already to be run from a separate thread than the CHIP thread (see calls to `WaitForResponse()` later on below).

If the application wants to pivot to running application logic on the CHIP thread (FYI - this function is part of the application), then it needs to alter the contract and logic of this function and consequently, take out the locking as needed.